libnmstate/netapplier.py

```diff
@@ -0,0 +1,49 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+#
+# Refer to the README and COPYING files for full details of the license
+#
+
+from libnmstate import nmclient
+from libnmstate import validator
+
+
+def apply(desired_state):
+    validator.verify(desired_state)
+
+    _apply_ifaces_state(desired_state)
+
+
+def _apply_ifaces_state(state):
+    client = nmclient.client()
+
+    for iface_state in state['interfaces']:
+        nmdev = client.get_device_by_iface(iface_state['name'])
+        if iface_state['state'] == 'up':
+            if nmdev.get_state() == nmclient.NM.DeviceState.ACTIVATED:
+                continue
+            client.activate_connection_async(device=nmdev)
```

Comment:
what if there is no connection for the interface? don't we need to create one?

Reply:
We assume it has one, which for the majority of the device types it is a valid assumption. We may need to change this assumption for a more generic solution. For now it is focused on the minimum PoC scenario we aim for.